tinyDA/distributions.py

```diff
@@ -207,6 +207,18 @@ def GaussianLogLike(data, covariance):
         IsotropicGaussianLogLike, depending on the structure of the
         giver covariance matrix.
     """
+
+    # check if covariance operator is a square numpy array.
+    if not isinstance(covariance, np.ndarray):
+        raise TypeError("Covariance must be a 2-D numpy array.")
+    elif covariance.ndim == 1:
```

Comment:
I prefer to keep the `if-elif` hierarchy a little flatter and only use `elif` if the cases are mutually exclusive, which makes it a little more readable to me. This line could be an `if` instead of an `elif`.